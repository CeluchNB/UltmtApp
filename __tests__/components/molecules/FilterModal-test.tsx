import FilterModal from '../../../src/components/molecules/FilterModal'
import React from 'react'
import { act, fireEvent, render } from '@testing-library/react-native'

import MockDate from 'mockdate'
MockDate.set('01 October 2022 00:00 UTC')

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

it('should match snapshot', () => {
    const snapshot = render(
        <FilterModal
            visible={true}
            defaultValues={{
                live: 'true',
                after: new Date(),
                before: new Date(),
            }}
            onClose={() => {}}
        />,
    )

    expect(snapshot).toMatchSnapshot()
})

it('should call on close with correct data', async () => {
    const onClose = jest.fn()
    const { getByText } = render(
        <FilterModal
            visible={true}
            defaultValues={{
                live: 'true',
                after: new Date('2022-01-01'),
                before: new Date('2022-06-01'),
            }}
            onClose={onClose}
        />,
    )

    const doneBtn = getByText('done')
    fireEvent.press(doneBtn)
    await act(async () => {})

    expect(onClose).toBeCalled()
})

// it('should handle clear press', async () => {
//     const onClose = jest.fn()
//     const { getByText, getAllByTestId, getByPlaceholderText } = render(
//         <FilterModal
//             visible={true}
//             defaultValues={{
//                 live: 'true',
//                 after: new Date('2022-01-01'),
//                 before: new Date('2022-06-01'),
//             }}
//             onClose={onClose}
//         />,
//     )

//     const textInput = getAllByTestId('date-text-input')
//     fireEvent.changeText(textInput[0], '1970-01-01')

//     const clearBtn = getByText('clear')
//     fireEvent.press(clearBtn)

//     expect(getByPlaceholderText('1/1/2022')).not.toBeNull()
// })

it('should close on request close', () => {
    const onClose = jest.fn()
    const { getByTestId } = render(
        <FilterModal
            visible={true}
            defaultValues={{
                live: 'true',
                after: new Date('2022-01-02'),
                before: new Date('2022-06-01'),
            }}
            onClose={onClose}
        />,
    )

    const modal = getByTestId('base-modal')
    fireEvent(modal, 'onRequestClose')

    expect(onClose).toHaveBeenCalled()
})
