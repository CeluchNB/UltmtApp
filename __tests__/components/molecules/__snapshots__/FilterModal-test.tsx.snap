// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="base-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#30303050",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#303030",
          "borderColor": "#ffd600",
          "borderRadius": 10,
          "color": "#ffd600",
          "elevation": 5,
          "flex": 1,
          "margin": 20,
          "padding": 25,
          "shadowColor": "#ffd600",
        }
      }
    >
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "flex": 1,
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            }
          }
        >
          Filters
        </Text>
        <View
          collapsable={false}
          style={
            {
              "alignSelf": undefined,
              "bottom": undefined,
              "end": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "start": undefined,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "#ffd600",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          {
                            "textTransform": "uppercase",
                          },
                          {
                            "color": "#ffd600",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    clear
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            accessibilityRole="radiogroup"
          >
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffd600",
                    "fontSize": 41.0958904109589,
                    "fontWeight": "bold",
                  }
                }
              >
                All Games
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "checked": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                color="#ffd600"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    {
                      "borderRadius": 18,
                      "padding": 6,
                    },
                  ]
                }
                uncheckedColor="#31adff"
              >
                <View
                  style={
                    {
                      "opacity": 0,
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#ffd600",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ó°„¬
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffd600",
                    "fontSize": 41.0958904109589,
                    "fontWeight": "bold",
                  }
                }
              >
                Live Games
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "checked": true,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                color="#ffd600"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    {
                      "borderRadius": 18,
                      "padding": 6,
                    },
                  ]
                }
                uncheckedColor="#31adff"
              >
                <View
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#ffd600",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ó°„¬
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffd600",
                    "fontSize": 41.0958904109589,
                    "fontWeight": "bold",
                  }
                }
              >
                Completed Games
              </Text>
              <View
                accessibilityLiveRegion="polite"
                accessibilityRole="radio"
                accessibilityState={
                  {
                    "checked": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                color="#ffd600"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    {
                      "borderRadius": 18,
                      "padding": 6,
                    },
                  ]
                }
                uncheckedColor="#31adff"
              >
                <View
                  style={
                    {
                      "opacity": 0,
                    }
                  }
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "#ffd600",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ó°„¬
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "color": "#ffd600",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                      },
                      {
                        "backgroundColor": "#303030",
                        "borderColor": "#ffd600",
                        "borderRadius": 4,
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                  testID="text-input-outline"
                />
                <View>
                  <View
                    style={
                      [
                        {
                          "paddingBottom": 0,
                        },
                        {
                          "minHeight": 56,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <TextInput
                      editable={true}
                      focusable={false}
                      maxFontSizeMultiplier={1.5}
                      multiline={false}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onPressOut={[Function]}
                      placeholder="10/1/2022"
                      placeholderTextColor="#ffd600"
                      selectionColor="#ffd600"
                      style={
                        [
                          {
                            "margin": 0,
                            "zIndex": 1,
                          },
                          {
                            "height": 48,
                          },
                          {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "letterSpacing": 0.15,
                            "lineHeight": undefined,
                            "paddingHorizontal": 16,
                            "textAlign": "left",
                            "textAlignVertical": "center",
                          },
                          false,
                          [
                            {},
                          ],
                          undefined,
                        ]
                      }
                      testID="date-text-input"
                      underlineColorAndroid="transparent"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Text
            style={
              [
                {
                  "color": "#ffd600",
                  "fontSize": 41.0958904109589,
                  "fontWeight": "bold",
                },
                {
                  "margin": 20,
                },
              ]
            }
          >
            to
          </Text>
          <View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  {
                    "color": "#ffd600",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    [
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                      },
                      {
                        "backgroundColor": "#303030",
                        "borderColor": "#ffd600",
                        "borderRadius": 4,
                        "borderWidth": 1,
                      },
                      undefined,
                    ]
                  }
                  testID="text-input-outline"
                />
                <View>
                  <View
                    style={
                      [
                        {
                          "paddingBottom": 0,
                        },
                        {
                          "minHeight": 56,
                          "paddingTop": 8,
                        },
                      ]
                    }
                  >
                    <TextInput
                      editable={true}
                      focusable={false}
                      maxFontSizeMultiplier={1.5}
                      multiline={false}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onPressOut={[Function]}
                      placeholder="10/1/2022"
                      placeholderTextColor="#ffd600"
                      selectionColor="#ffd600"
                      style={
                        [
                          {
                            "margin": 0,
                            "zIndex": 1,
                          },
                          {
                            "height": 48,
                          },
                          {
                            "paddingBottom": 0,
                            "paddingTop": 0,
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": undefined,
                            "letterSpacing": 0.15,
                            "lineHeight": undefined,
                            "paddingHorizontal": 16,
                            "textAlign": "left",
                            "textAlignVertical": "center",
                          },
                          false,
                          [
                            {},
                          ],
                          undefined,
                        ]
                      }
                      testID="date-text-input"
                      underlineColorAndroid="transparent"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignSelf": undefined,
            "bottom": undefined,
            "end": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "start": undefined,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#ffd600",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 0,
                "minWidth": "auto",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  false,
                  {
                    "borderRadius": 20,
                  },
                ]
              }
              testID="button"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 24,
                          "marginVertical": 10,
                        },
                        {
                          "marginHorizontal": 8,
                        },
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  done
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
